morecantile/models.py

```diff
@@ -45,3 +93,278 @@ class TileMatrixSet(BaseModel):
     def crs(self) -> CRS:
         """Fetch CRS from epsg"""
         return CRS.from_epsg(self.supportedCRS.split("/")[-1])
+
+    @property
+    def meters_per_unit(self) -> float:
+        """
+        coefficient to convert the coordinate reference system (CRS)
+        units into meters (metersPerUnit).
+
+        From note g in http://docs.opengeospatial.org/is/17-083r2/17-083r2.html#table_2:
+          If the CRS uses meters as units of measure for the horizontal dimensions,
+          then metersPerUnit=1; if it has degrees, then metersPerUnit=2pa/360
+          (a is the Earth maximum radius of the ellipsoid).
+
+        """
+        # self.crs.linear_units_factor[1]  GDAL 3.0
+        return (
+            1.0 if self.crs.linear_units == "metre" else 2 * math.pi * 6378137 / 360.0
+        )
+
+    @classmethod
+    def load(cls, name: str):
+        """Load default TileMatrixSet."""
+        try:
+            return cls.parse_file(os.path.join(data_dir, f"{name}.json"))
+        except FileNotFoundError:
+            raise NotAValidName(f"'{name}' is not a valid default TileMatrixSet.")
+
+    @classmethod
+    def custom(
+        cls,
+        extent: List[float],
+        crs: CRS,
+        resolution: float,
+        tile_width: int = 256,
+        tile_height: int = 256,
+        matrix_scale: List = [1, 1],
+    ):
+        """
+        Construct a custom TileMatrixSet.
+
+        Attributes
+        ----------
+        crs: rasterio.crs.CRS
+            TileMatrixSet coordinate reference system
+        extent: list
+            Bounding box of the TileMatrixSet
+        resolution: float
+        matrix_scale: list
+            Tiling schema coalescence coefficient (default: [1, 1] for EPSG:3857).
+            Should be set to [2, 1] for EPSG:4326.
+            see: http://docs.opengeospatial.org/is/17-083r2/17-083r2.html#14
+
+        Returns:
+        --------
+        TileMatrixSet
+
+        """
+        raise NotImplementedError
+
+    def matrix(self, zoom: int) -> TileMatrix:
+        """Return the TileMatrix for a specific zoom."""
+        zoom_str = str(zoom)
+        for matrix in self.tileMatrix:
+            if matrix.identifier == zoom_str:
+                return matrix
+
+        raise Exception(f"TileMatrix not found for level: {zoom}")
+
+    def _resolution(self, matrix: TileMatrix) -> float:
```

Comment:
From the docstring I would think this takes input of `zoom`

```python
return self.matrix(zoom) * 0.28e-3 / self.meters_per_unit
```

Reply:
I edited the Docstring to 
> Tile resolution for a TileMatrix.

We could use `zoom` but then we will have to find the coresponding tileMatrix which is already done in most function calling `_resolution`